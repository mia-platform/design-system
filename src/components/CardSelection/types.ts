
export enum Type {
  Radio = 'radio',
  Checkbox = 'checkbox'
}
