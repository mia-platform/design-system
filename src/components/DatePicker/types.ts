import {
  DisabledDate as AntDisabledDate,
} from 'rc-picker/lib/interface'
import { Dayjs } from 'dayjs'

export type DisabledDate = AntDisabledDate<Dayjs>
export enum ShowTimeOptions {
  Hours = 'hours',
  Minutes = 'minutes',
  Seconds = 'seconds'
}

export type ShowTime = boolean | {
  showHour: boolean;
  showMinute: boolean;
  showSecond: boolean;
}
