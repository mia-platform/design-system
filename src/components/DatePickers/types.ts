/**
 * Copyright 2025 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  DisabledDate as AntDisabledDate,
} from 'rc-picker/lib/interface'
import { Dayjs } from 'dayjs'

export type DisabledDate = AntDisabledDate<Dayjs>
export enum ShowTimeOptions {
  Hours = 'hours',
  Minutes = 'minutes',
  Seconds = 'seconds'
}

export type AntShowTimeOptions = boolean | {
  showHour: boolean;
  showMinute: boolean;
  showSecond: boolean;
}

export type NoUndefinedRangeValueType = [start: Dayjs | null, end: Dayjs | null];

export type RangeValueType = [
  start: Dayjs | null | undefined,
  end: Dayjs | null | undefined
];

export const defaultTimeFormat = 'HH:mm'
export const defaultDateFormat = 'DD/MM/YYYY'
