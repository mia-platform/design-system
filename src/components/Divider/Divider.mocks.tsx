/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, ReactNode } from 'react'

type SplitTextComponentProps = {
  children: ReactNode,
  mockedTextEntries?: string[],
}

const getTextComponent = (value: string | undefined): ReactNode => (<span key={`item-${value}`}>{value}</span>)

export const LOREM_IPSUM = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'

export const SeparateTextComponent = ({
  children,
  mockedTextEntries,
}: SplitTextComponentProps): ReactElement => {
  const defaultTextEntries = ['Text mocked 1', 'Text mocked 2']
  const textEntries = mockedTextEntries || defaultTextEntries

  const separatedComponents = textEntries
    .reduce((acc: ReactNode[], value, i) => {
      if (i === textEntries.length - 1) {
        return [
          ...acc,
          getTextComponent(textEntries.at(textEntries.length - 1)),
        ]
      }

      return [
        ...acc,
        getTextComponent(value),
        children,
      ]
    }, [])

  return (
    // eslint-disable-next-line react/jsx-no-useless-fragment
    <>
      {separatedComponents}
    </>
  )
}
