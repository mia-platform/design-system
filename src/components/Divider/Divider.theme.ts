/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

// import { ComponentsTheme } from '../ThemeProvider/Ant'
// import Theme from '../../themes/schema'

/**
 * Generates a Ant theme configuration for Button component based on a theme configuration.
 *
 * @link https://ant.design/components/divider#design-token
 *
 * @param {Partial<Theme>} theme - theme configuration.
 * @returns {Partial<ComponentsTheme>} The generated Divider Ant theme configuration.
 */


// FIXME: Could theme affect the props of this component?
// export default ({ typography }: Partial<Theme>): ComponentsTheme['Divider'] => ({
//   orientationMargin: 0,
//   textPaddingInline: '?',
//   verticalMarginInline: '?',
// })
