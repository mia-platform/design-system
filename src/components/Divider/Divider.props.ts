/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'

import { TextOrientation, Type } from './Divider.types'

export type DividerProps = {

  /**
   * The wrapped title to be shown in the divider.
   */
  children?: ReactNode,

  /**
   * Show the divider as dashed line. Default is false.
   */
  isDashed?: boolean,

  /**
   * The margin-left/right between the title and its closest border, based on the orientation left or right.
   * If a numeric value of type string is provided without a unit, it is assumed to be in pixels (px) by default.
   */
  orientationMargin?: string | number,

  /**
   * Defines the text orientation on the divider. Either:
   *
   * - center(default)
   * - left
   * - right
   */
  orientation?: TextOrientation,

  /**
   * Show the text on divider as plain style. Default is true.
   */
  isPlain?: boolean,

  /**
   * Defines the divider type. Either:
   *
   * - horizontal(default)
   * - vertical
   */
  type?: Type
}
