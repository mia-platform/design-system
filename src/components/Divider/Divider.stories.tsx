/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import type { Meta, StoryObj } from '@storybook/react'

import { LOREM_IPSUM, SeparateTextComponent } from './Divider.mocks'
import { Divider } from '.'
import { Orientation } from './Divider.types'

const meta = {
  args: {
    ...Divider.defaultProps,
  },
  argTypes: {
    text: {
      control: {
        type: 'text',
      },
    },
  },
  component: Divider,
} satisfies Meta<typeof Divider>

export default meta
type Story = StoryObj<typeof meta>

export const Horizontal: Story = {
  render: (args) => (
    <SeparateTextComponent
      mockedTextEntries={[LOREM_IPSUM, LOREM_IPSUM]}
    >
      <Divider {...args} />
    </SeparateTextComponent>
  ),
}

export const Vertical: Story = {
  render: () => (
    <SeparateTextComponent>
      <Divider orientation={Orientation.Vertical} />
    </SeparateTextComponent>
  ),
}

export const WithTitle: Story = {
  render: () => (
    <SeparateTextComponent
      mockedTextEntries={[LOREM_IPSUM, LOREM_IPSUM]}
    >
      <Divider text="Some text" />
    </SeparateTextComponent>
  ),
}
