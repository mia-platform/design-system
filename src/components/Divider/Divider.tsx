/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Divider as AntDivider } from 'antd'
import { ReactElement } from 'react'

import { Orientation, TextOrientation } from './Divider.types'
import { DividerProps } from './Divider.props'
import { useTheme } from '../../hooks/useTheme'

const { Horizontal } = Orientation
const { Left } = TextOrientation

/**
 * A divider line to separate different content
 *
 * @link https://ant.design/components/divider
 * @returns {Divider} Divider component
 */
export const Divider = ({
  orientation,
  text,
  textOrientation,
}: DividerProps): ReactElement => {
  const { spacing } = useTheme()

  return (
    <AntDivider
      orientation={textOrientation}
      orientationMargin={spacing?.margin?.xl}
      type={orientation}
    >
      {text}
    </AntDivider>
  )
}

Divider.defaultProps = {
  orientation: Horizontal,
  textOrientation: Left,
}
