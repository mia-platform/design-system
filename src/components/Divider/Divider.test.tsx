/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Divider } from '.'
import { Orientation } from './Divider.types'
import { SeparateTextComponent } from './Divider.mocks'
import { render } from '../../test-utils'

describe('Divider Component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders divider', () => {
    const { asFragment } = render(
      <SeparateTextComponent>
        <Divider />
      </SeparateTextComponent>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders horizontal divider with title correctly', () => {
    const { asFragment } = render(
      <SeparateTextComponent>
        <Divider text="Some text" />
      </SeparateTextComponent>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders vertical divider correctly', () => {
    const { asFragment } = render(
      <SeparateTextComponent>
        <Divider orientation={Orientation.Vertical} />
      </SeparateTextComponent>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
