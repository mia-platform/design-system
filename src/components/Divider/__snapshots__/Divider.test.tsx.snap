// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component renders divider 1`] = `
<DocumentFragment>
  <span>
    Text mocked 1
  </span>
  <div
    class="ant-divider ant-divider-horizontal ant-divider-no-default-orientation-margin-left"
    role="separator"
  />
  <span>
    Text mocked 2
  </span>
</DocumentFragment>
`;

exports[`Divider Component renders horizontal divider with title correctly 1`] = `
<DocumentFragment>
  <span>
    Text mocked 1
  </span>
  <div
    class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left ant-divider-no-default-orientation-margin-left"
    role="separator"
  >
    <span
      class="ant-divider-inner-text"
      style="margin-left: 24px;"
    >
      Some text
    </span>
  </div>
  <span>
    Text mocked 2
  </span>
</DocumentFragment>
`;

exports[`Divider Component renders vertical divider correctly 1`] = `
<DocumentFragment>
  <span>
    Text mocked 1
  </span>
  <div
    class="ant-divider ant-divider-vertical ant-divider-no-default-orientation-margin-left"
    role="separator"
  />
  <span>
    Text mocked 2
  </span>
</DocumentFragment>
`;
